test/test_sendfile.rs

```diff
@@ -23,7 +24,12 @@ fn test_sendfile_linux() {
 
     let (rd, wr) = pipe().unwrap();
     let mut offset: off_t = 5;
-    let res = sendfile(wr, tmp.as_raw_fd(), Some(&mut offset), 2).unwrap();
+    // The construct of this `OwnedFd` is a temporary workaround, when `pipe(2)`
+    // becomes I/O-safe:
+    // pub fn pipe() -> std::result::Result<(OwnedFd, OwnedFd), Error>
+    // then it is no longer needed.
+    let wr = unsafe { OwnedFd::from_raw_fd(wr) };
+    let res = sendfile(&wr, &tmp, Some(&mut offset), 2).unwrap();
```

Comment:
This is a temporary workaround to cope with the existing `pipe(2)` interface and should be resolved in other PRs.